### PR TITLE
Flask-APP

### PR DESCRIPTION
**Pull request by Yue-Fu1**
Basic architecture about flask, including stations and bike details

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -1,25 +1,159 @@
...
+@app.errorhandler(404)
+@cross_origin()
+def page_not_found(error):
+    return flask.render_template('404.html'), 404
+
+@app.route("/stations")
+@cross_origin()
 def get_stations():
-    db_utils.get_stations()
+    engine = scraper.connect_db("ubuntu@ec2-18-206-204-211.compute-1.amazonaws.com:22", "3306", " dublin-bikes", "admin", "0d72d379765757bc1")
```
Comment (AprilPolubiec) [inline app.py:27]: hi @Yue-Fu1 - this is confidential information! you should be importing this data from the `secure/credentials.json`- see an example in db_utils.py

File: app.py [hunk 2]
```diff
@@ -1,25 +1,159 @@
...
+    sql = "select * from bike_stations;"
+    rows = engine.execute(sql).fetchall()
+    print("#found {} stations", len(rows))
+    stations = jsonify(stations=[dict(row) for row in rows])    
+    engine.dispose()
+    return stations
```
Comment (AprilPolubiec) [inline app.py:33]: this work looks great but I think I already did all of this in `db_utils` - was there a reason why you chose to remove these? Example:

```
def get_stations():
    db_utils.get_stations()
```

Maybe it was just a mistake, which is okay!

File: app.py [hunk 3]
```diff
@@ -1,25 +1,159 @@
...
+    return availability
+
+@app.route('/station_details', methods=['GET', 'POST'])
+@cross_origin()
+def station_details():
+    """Function to get dyanmic details for stations"""
+    #Info will be pulled from a javascript function on the home page
+    station_number = request.args.get('station_number')
+    engine = scraper.connect_db("ubuntu@ec2-18-206-204-211.compute-1.amazonaws.com:22", "3306", " dublin-bikes", "admin", "0d72d379765757bc1")
+    sql = "SELECT *, station_name FROM availability, bike_stations WHERE availability.station_number = %s and availability.station_number = bike_stations.station_number ORDER BY last_updated DESC LIMIT 1;"
```
Comment (AprilPolubiec) [inline app.py:54]: Does this code work? Because these columns are not in our database

**Comment by AprilPolubiec** [comment]
The code looks great but it duplicates some functions that already exist. It also is not calling the correct columns from the database - you can see all of the column names in db_utils. But overall great job you clearly did a lot of work and learned more about flask 👏 👏

File: app.py (hunk 1 above)
Comment (aidacoughlan) [reply app.py:27]: @Yue-Fu1  and make sure your secure file is up to date with the extra keys in the discord.

**Pull request closed without merging**